velox/exec/HashProbe.cpp

```diff
@@ -998,19 +998,28 @@ RowVectorPtr HashProbe::getOutputInternal(bool toSpillOutput) {
       ? inputSize
       : outputBatchSize_;
   outputTableRowsCapacity_ = outputBatchSize;
-  if (filter_ &&
-      (isLeftJoin(joinType_) || isFullJoin(joinType_) ||
-       isAntiJoin(joinType_))) {
-    // If we need non-matching probe side row, there is a possibility that such
-    // row exists at end of an input batch and being carried over in the next
-    // output batch, so we need to make extra room of one row in output.
-    ++outputTableRowsCapacity_;
+  if (filter_) {
+    if (isLeftJoin(joinType_) || isFullJoin(joinType_) ||
+        isAntiJoin(joinType_)) {
+      // If we need non-matching probe side row, there is a possibility that
+      // such row exists at end of an input batch and being carried over in the
+      // next output batch, so we need to make extra room of one row in output.
+      ++outputTableRowsCapacity_;
+    }
+
+    // Intialize 'leftSemiProjectIsNull_' for null aware lft semi join.
+    if (isLeftSemiProjectJoin(joinType_) && nullAware_) {
+      leftSemiProjectIsNull_.clearAll();
```

Comment:
Can we do `leftSemiProjectIsNull_.resize(outputTableRowsCapacity_)` here so we don't need to do it in each loop?  `SelectivityVector::resize` requires iterate over all the bits in it.

Reply:
Looks like we are still resizing it in `evalFilter`?

Reply:
Moved it outside the loop, thanks.